scripts/install-appstudio-kcp.sh

```diff
@@ -0,0 +1,308 @@
+        git clone https://github.com/redhat-appstudio/infra-deployments "$WORKSPACE"/tmp/infra-deployments
+    fi
+}
+
+# Create preview.env file for App Studio installation. More info about this can be found in the redhat-appstudio/infra-deployments repo
+function createPreviewEnvFile() {
+    cat > "$WORKSPACE"/tmp/infra-deployments/hack/preview.env << EOF
+export CLUSTER_KUBECONFIG="$CLUSTER_KUBECONFIG"
+export KCP_KUBECONFIG="$KCP_KUBECONFIG"
+export ROOT_WORKSPACE="$APPSTUDIO_ROOT"
+export APPSTUDIO_WORKSPACE="redhat-appstudio-${WORKSPACE_ID}"
+export HACBS_WORKSPACE="redhat-hacbs-${WORKSPACE_ID}"
+export USER_APPSTUDIO_WORKSPACE="appstudio-${WORKSPACE_ID}"
+export COMPUTE_WORKSPACE="compute-${WORKSPACE_ID}"
+export USER_HACBS_WORKSPACE="appstudio-${WORKSPACE_ID}"
+EOF
+}
+
+# Add a custom remote for infra-deployments repo and start the installation
+function startRedHatAppStudioInstallation() {
+    # If we are in infra-deployments jobs we don't need to clone infra-deployments. Openshift CI clones automatically
+    if [[ "$JOB_TYPE" == "periodic" ]] || [[ "$REPO_NAME" == "infra-deployments" ]]
+    then
+        git remote add "${MY_GIT_FORK_REMOTE}" https://github.com/"${MY_GITHUB_ORG}"/infra-deployments.git
+        "$WORKSPACE"/hack/bootstrap.sh -m preview
+    else
+        cd "$WORKSPACE"/tmp/infra-deployments
+        git remote add "${MY_GIT_FORK_REMOTE}" https://github.com/"${MY_GITHUB_ORG}"/infra-deployments.git
+        "$WORKSPACE"/tmp/infra-deployments/hack/bootstrap.sh -m preview
+    fi
+}
+
+# Start to run the tests
+function runE2Etests() {
+    # If we are in infra-deployments jobs we don't need to clone infra-deployments. Openshift CI clones automatically
+    if [[ "$REPO_NAME" == "e2e-tests" ]]
+    then
+        make local/cluster/prepare
+        make local/test/e2e
+    else
+        if [ -d "$WORKSPACE"/tmp/e2e-tests ] 
+        then
+            echo -e "[WARN] tmp/e2e-tests already exists. Deleting..." 
+            rm -rf "$WORKSPACE""tmp/e2e-tests"
+        fi
+        git clone -b kcp_scr https://github.com/flacatus/e2e-tests.git "$WORKSPACE""tmp/e2e-tests"
+        cd "$WORKSPACE""tmp/e2e-tests"
+
+        make local/cluster/prepare
+        make local/test/e2e
+        cd "$WORKSPACE"
+    fi
+}
+
+function runAuthenticationInBackground() {
+    export -f redHatSSOAuthentication
+    timeout --kill-after=110m --foreground 100m bash -c -- 'while [ true ]; do redHatSSOAuthentication; sleep 60s; done'
+}
+
+# Run the authentication SSO only if we are in Openshift CI.
+if [[ $CI == "true" ]]
+then
+    runAuthenticationInBackground &
+    BG_PID=$!
+    TIMEOUT_PID=$(ps -o pid= --ppid="${BG_PID}")
+fi
+
+installKubectlOIDCLoginPlugin
+installKubectlKcpPlugins
+cloneInfraDeployments
+createAppStudioRootWorkspace
+createPreviewEnvFile
+startRedHatAppStudioInstallation
+
+# Switch to user workspace
+kubectl ws "${USER_APPSTUDIO_WORKSPACE}"
+
+if [[ $RUN_E2E == "true" ]]
```

Comment:
One more thing. I don't understand why running tests is part of the "install-appstudio-kcp" script.

Wouldn't it make sense to wrap this script by `ci/test/e2e` implemented in magefile and move [this part](https://github.com/redhat-appstudio/e2e-tests/pull/207/files#diff-2e8bf596f11e4dfebfd3ad4dc7109218463dbe94fb89885607c79d1717f45072R261-R280) to [RunE2ETests function](https://github.com/redhat-appstudio/e2e-tests/blob/ce300a7389d7e370004954fc9ec95c68d8d8b684/magefiles/magefile.go#L137)?